src/libs/actions/TaxRate.ts

```diff
@@ -463,6 +483,81 @@ function renamePolicyTax(policyID: string, taxID: string, newName: string) {
     API.write(WRITE_COMMANDS.RENAME_POLICY_TAX, parameters, onyxData);
 }
 
+function setPolicyTaxCode(policyID: string, oldTaxCode: string, newTaxCode: string) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const originalTaxRate = {...policy?.taxRates?.taxes[oldTaxCode]};
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [oldTaxCode]: null,
+                            [newTaxCode]: {
+                                ...originalTaxRate,
+                                pendingFields: {code: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                                errorFields: {code: null},
+                                previousTaxCode: oldTaxCode,
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [oldTaxCode]: null,
+                            [newTaxCode]: {
+                                ...originalTaxRate,
+                                code: newTaxCode,
+                                pendingFields: {code: null},
+                                pendingAction: null,
+                                errorFields: {code: null},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [newTaxCode]: null,
+                            [oldTaxCode]: {
+                                ...originalTaxRate,
+                                code: originalTaxRate.code,
+                                pendingFields: {code: null},
+                                pendingAction: null,
+                                errorFields: {code: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('workspace.taxes.error.updateFailureMessage')},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters: UpdatePolicyTaxCodeParams = {
+        policyID,
```

Comment:
You'll need to pass the taxID (the name) here. We only really use it if the newTaxCode is clear, but you might s well pass it always, for simplicity

Reply:
taxCode is same as taxID right? 

do you mean that the api expects?
taxID: newTaxCode

Reply:
i.e. just a different variable name

Reply:
Ah, not quite. It expects the `name`, which is often similar (as we get the default code based on the name, which is why we need it when empty), but not  the same as the taxCode.

This one:
<img width="159" alt="Screenshot 2024-07-15 at 8 51 39 AM" src="https://github.com/user-attachments/assets/5216ffc5-c97a-4aca-a5d0-e3e9bd58442d">